test/contracts/OVM/bridge/assets/OVM_L1ERC20Gateway.spec.ts

```diff
@@ -0,0 +1,232 @@
+import { expect } from '../../../../setup'
+
+/* External Imports */
+import { ethers } from 'hardhat'
+import { Signer, ContractFactory, Contract, BigNumber } from 'ethers'
+import {
+  smockit,
+  MockContract,
+  smoddit,
+  ModifiableContract,
+} from '@eth-optimism/smock'
+
+/* Internal Imports */
+import { NON_ZERO_ADDRESS, ZERO_ADDRESS } from '../../../../helpers'
+
+const HARDCODED_GASLIMIT = 8999999
```

Comment:
Probably good to resolve this based on the ExecutionManager. Of course ideally we move the env mgr